pkg/provider/securitygroup/securitygroup.go

```diff
@@ -319,12 +319,50 @@ func (helper *RuleHelper) RemoveDestinationFromRules(
 
 		if err := helper.removeDestinationFromRule(rule, dstPrefixes, retainDstPorts); err != nil {
 			logger.Error(err, "Failed to remove destination from rule", "rule-name", *rule.Name)
+			return err
 		}
 	}
 
 	return nil
 }
 
+// RetainDestinationFromRules retains the given destination prefixes from the rules.
+func (helper *RuleHelper) RetainDestinationFromRules(dstPrefixes []string) error {
+	logger := helper.logger.WithName("RetainDestinationFromRules").WithValues("num-dst-prefixes", len(dstPrefixes))
+	logger.V(10).Info("Start retaining")
+	defer logger.V(10).Info("Completed retaining")
+
+	for _, rule := range helper.rules {
+		if rule.Properties.Priority == nil {
+			continue
+		}
+		priority := *rule.Properties.Priority
+		if priority < consts.LoadBalancerMinimumPriority || consts.LoadBalancerMaximumPriority < priority {
```

Comment:
we are unable to prevent users to use this priority ranges, so user rules may be included here. Could you also filter rule name here?